yb-voyager/src/queryissue/unsupported_dml_constructs.go

```diff
@@ -111,3 +111,117 @@ func (d *XmlExprDetector) Detect(msg protoreflect.Message) ([]string, error) {
+
+ASSUMPTION:
+- RangeTableFunc is used for representing XMLTABLE() only as of now
+- Comments from Postgres code:
+  - RangeTableFunc - raw form of "table functions" such as XMLTABLE
+  - Note: JSON_TABLE is also a "table function", but it uses JsonTable node,
+  - not RangeTableFunc.
+
+- link: https://github.com/postgres/postgres/blob/ea792bfd93ab8ad4ef4e3d1a741b8595db143677/src/include/nodes/parsenodes.h#L651
+*/
+type RangeTableFuncDetector struct{}
+
+func NewRangeTableFuncDetector() *RangeTableFuncDetector {
+	return &RangeTableFuncDetector{}
+}
+
+// Detect checks if a RangeTableFunc node is present for a XMLTABLE() function
+func (d *RangeTableFuncDetector) Detect(msg protoreflect.Message) ([]string, error) {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_RANGETABLEFUNC_NODE {
+		if isXMLTable(msg) {
+			return []string{XML_FUNCTIONS}, nil
+		}
+	}
+	return nil, nil
+}
+
+/*
+Note: XMLTABLE() is not a simple function(stored in FuncCall node), its a table function
+which generates set of rows using the info(about rows, columns, content) provided to it
+Hence its requires a more complex node structure(RangeTableFunc node) to represent.
+
+XMLTABLE transforms XML data into relational table format, making it easier to query XML structures.
+Detection in RangeTableFunc Node:
+- docexpr: Refers to the XML data source, usually a column storing XML.
+- rowexpr: XPath expression (starting with '/' or '//') defining the rows in the XML.
+- columns: Specifies the data extraction from XML into relational columns.
+
+Example: Converting XML data about books into a table:
+SQL Query:
+
+	SELECT x.*
+	FROM XMLTABLE(
+		'/bookstore/book'
+		PASSING xml_column
+		COLUMNS
+			title TEXT PATH 'title',
+			author TEXT PATH 'author'
+	) AS x;
+
+Parsetree: stmt:{select_stmt:{target_list:{res_target:{val:{column_ref:{fields:{string:{sval:"x"}}  fields:{a_star:{}}  location:7}}  location:7}}
+from_clause:{range_table_func:
+	{docexpr:{column_ref:{fields:{string:{sval:"xml_column"}}  location:57}}
+	rowexpr:{a_const:{sval:{sval:"/bookstore/book"}  location:29}}
+	columns:{range_table_func_col:{colname:"title"  type_name:{names:{string:{sval:"text"}}  typemod:-1  location:87}  colexpr:{a_const:{sval:{sval:"title"}  location:97}}  location:81}}
+	columns:{range_table_func_col:{colname:"author"  type_name:{names:{string:{sval:"text"}}  typemod:-1  location:116}  colexpr:{a_const:{sval:{sval:"author"}  location:126}}  location:109}}
+alias:{aliasname:"x"}  location:17}}  limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}  stmt_len:142
+
+Here, 'docexpr' points to 'xml_column' containing XML data, 'rowexpr' selects each 'book' node, and 'columns' extract 'title' and 'author' from each book.
+Hence Presence of XPath in 'rowexpr' and structured 'columns' typically indicates XMLTABLE usage.
+*/
+// Function to detect if a RangeTableFunc node represents XMLTABLE()
+func isXMLTable(rangeTableFunc protoreflect.Message) bool {
```

Comment:
If you pass in a message that is NOT a rangeTableFunc to this function, ensure that it returns false. An explicit validation that msg is a rangeTableFunc might also help..

Reply:
currently the detector has that check. 
```
// Detect checks if a RangeTableFunc node is present for a XMLTABLE() function
func (d *RangeTableFuncDetector) Detect(msg protoreflect.Message) ([]string, error) {
	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_RANGETABLEFUNC_NODE {
		if isXMLTable(msg) {
			return []string{XML_FUNCTIONS}, nil
		}
	}
	return nil, nil
}
```

Do you still feel the need to add a check here, considering func arg name is `rangeTableFunc protoreflect.Message`?

Reply:
Ah, I see. 
The way I see it is that if you're defining a function called `IsXMLTable(protoreflect.Message)`, it should be able to handle any input of protoreflect.Message and return the proper result. 

If you define it as `isXMLTable(node rangeFuncNode)`, then the type is fixed to be rangeFuncNode, so that's okay. 
But currently, it's defined as `isXMLTable(rangeTableFunc protoreflect.Message)` and you're just making an assumption that the message is of type rangeTableFunc, which is not ideal. 
I would suggest: Either 
1. Don't make that assumption. Move the check of `GetMsgFullName(msg) == queryparser.PG_QUERY_RANGETABLEFUNC_NODE` inside `isXMLTable` 
OR 
2. define it as `isXMLTable(node rangeFuncNode)`. 

I don't think 2 is possible, given that you're dealing in proto messages, so maybe 1?

Reply:
added check in IsXMLTable() to make it independent also.